src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -0,0 +1,429 @@
+        ResultSet rs = null;
+        // Run DatabaseMetadata function if it exists, takes precedence over query
+        System.out.println("Running test: " + testEntry.description);
+        try {
+            if (testEntry.meta_function == null) {
+                rs = executeQuery(testEntry, conn);
+            } else {
+                rs = executeDBMetadataCommand(testEntry, conn.getMetaData());
+            }
+            assertNotEquals(rs, null);
+            if (generate) {
+                TestGenerator.generateBaselineTestFiles(testEntry, rs);
+            } else {
+                if (testEntry.ordered != null && testEntry.ordered) {
+                    validateResultsOrdered(testEntry, rs);
+                } else {
+                    validateResultsUnordered(testEntry, rs);
+                }
+                validateResultSetMetadata(testEntry, rs.getMetaData());
+            }
+        } finally {
+            if (rs != null) {
+                Statement statement = rs.getStatement();
+                if (statement != null) {
+                    statement.close();
+                }
+            }
+        }
+    }
+
+    private static ResultSet executeQuery(TestEntry entry, Connection conn) throws SQLException {
+        Statement stmt = conn.createStatement();
+        return stmt.executeQuery(entry.sql);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static ResultSet executeDBMetadataCommand(
+            TestEntry entry, DatabaseMetaData databaseMetaData)
+            throws SQLException, InvocationTargetException, IllegalAccessException {
+
+        List<Object> metadataFunction = entry.meta_function;
+        assertTrue(metadataFunction != null && metadataFunction.size() > 0);
+        String functionName = (String) metadataFunction.remove(0);
+        Method[] m = DatabaseMetaData.class.getMethods();
+
+        for (Method method : m) {
+            if (method.getName().equalsIgnoreCase(functionName)) {
+                Class<?>[] types = method.getParameterTypes();
+                if (method.getReturnType() != ResultSet.class) {
+                    throw new IllegalArgumentException(
+                            "expected: "
+                                    + ResultSet.class.getName()
+                                    + " found: "
+                                    + method.getReturnType());
+                }
+                if (method.getParameterCount() != metadataFunction.size()) {
+                    throw new IllegalArgumentException(
+                            "expected parameter count: "
+                                    + method.getParameterCount()
+                                    + " found: "
+                                    + metadataFunction.size());
+                }
+                if (method.getParameterCount() == 0) {
+                    return (ResultSet) method.invoke(databaseMetaData);
+
+                } else {
```

Comment:
```suggestion
                }
```
[nit] Since we `return` in the `if` block, let's drop the `else` here and move everything over one level of indentation.